usage/test-framework/src/main/java/org/apache/brooklyn/test/framework/SimpleCommandImpl.java

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * Implementation for {@link SimpleCommand}.
+ */
+public class SimpleCommandImpl extends AbstractEntity
+        implements SimpleCommand, DriverDependentEntity<SimpleCommandDriver> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SimpleCommandImpl.class);
+    private static final int A_LINE = 80;
+    private transient SimpleCommandDriver driver;
+
+    private Collection<? extends Location> locations;
+
+    @Override
+    public SimpleCommandDriver getDriver() {
+        return driver;
+    }
+
+    @Override
+    public Class<SimpleCommandDriver> getDriverInterface() {
+        return SimpleCommandDriver.class;
+    }
+
+    /**
+     * Gives the opportunity to sub-classes to do additional work based on the result of the command.
+     */
+    protected void handle(SimpleCommandDriver.Result result) {
+        LOG.debug("Result is {}\nwith output [\n{}\n] and error [\n{}\n]", new Object[] {
+                result.getExitCode(), shorten(result.getStdout()), shorten(result.getStderr())
+        });
+    }
+
+    private String shorten(String text) {
+        return Strings.maxlenWithEllipsis(text, A_LINE);
+    }
+
+    /**
+     * Does nothing in this class but gives sub-classes the opportunity to filter locations according to some criterion.
+     */
+    public Collection<? extends Location> filterLocations(Collection<? extends Location> locations) {
+        return locations;
+    }
+
+
+    @Override
+    public void init() {
+        super.init();
+        getLifecycleEffectorTasks().attachLifecycleEffectors(this);
+    }
+
+
+    protected void initDriver(MachineLocation machine) {
+        LOG.debug("Initializing simple command driver");
+        SimpleCommandDriver newDriver = doInitDriver(machine);
+        if (newDriver == null) {
+            throw new UnsupportedOperationException("cannot start "+this+" on "+machine+": no driver available");
+        }
+        driver = newDriver;
+    }
+
+    protected SimpleCommandDriver doInitDriver(MachineLocation machine) {
+        if (driver!=null) {
+            if (machine.equals(driver.getLocation())) {
+                return driver; //just reuse
+            } else {
+                LOG.warn("driver/location change is untested for {} at {}; changing driver and continuing", this, machine);
+                return newDriver(machine);
+            }
+        } else {
+            return newDriver(machine);
+        }
+    }
+
+    protected SimpleCommandDriver newDriver(MachineLocation machine) {
+        LOG.debug("Creating new simple command driver for {} from management context", machine);
+        EntityDriverManager entityDriverManager = getManagementContext().getEntityDriverManager();
+        return entityDriverManager.build(this, machine);
+    }
+
+    @Override
+    public void start(@EffectorParam(name = "locations") Collection<? extends Location> locations) {
```

Comment:
Don't need the `@EffectorParam` annotations on the concrete  class - just on the interface where it's declared as an effector.